server/weather.py

```diff
@@ -0,0 +1,23 @@
+from server import app
+from .base import cached_route
+import requests
+import datetime
+import os
+
+
+@app.route('/weather/', methods=['GET'])
+def retrieve_weather_data():
+    '''Retrieves the current weather from the Open Weather Map API.
+    Stores data in a cache whenever data is retrieved; cache is updated
+    if it has not been updated within 10 minutes.
+    '''
+    APPID = os.getenv('APPID')
```

Comment:
Can this be something more specific? Something like "OWM_API_KEY". Naming is hard.